app/src/main/java/ch/epfl/polychef/Firebase.java

```diff
@@ -0,0 +1,118 @@
+package ch.epfl.polychef;
+
+import android.util.Log;
+
+import com.google.firebase.database.DataSnapshot;
+import com.google.firebase.database.DatabaseError;
+import com.google.firebase.database.DatabaseReference;
+import com.google.firebase.database.FirebaseDatabase;
+import com.google.firebase.database.ValueEventListener;
+
+import java.util.UUID;
+import java.util.function.Function;
+
+import ch.epfl.polychef.recipe.Recipe;
+
+public abstract class Firebase {
+    private static final String TAG="Firebase";
+    private static int id;
+    //Get an instance of the firebase
+    public static FirebaseDatabase firebaseInstance=FirebaseDatabase.getInstance();
+
+    public static void addRecipeToFirebase(Recipe recipe){
+        Preconditions.checkArgument(recipe!=null);
+
+        DatabaseReference idRef = firebaseInstance.getReference("id");
+        //Get the last ID used in the database
+        idRef.addListenerForSingleValueEvent(new ValueEventListener() {
+            @Override
+            public void onDataChange(DataSnapshot dataSnapshot) {
+                // This method is called once with the initial value
+                id=dataSnapshot.getValue(Integer.class);
+            }
+
+            @Override
+            public void onCancelled(DatabaseError error) {
+                // Failed to read value
+                Log.w(TAG, "Failed to read value.", error.toException());
+            }
+        });
+        //Change the value of the ID in the database
+        id+=1;
+        idRef.setValue(id);
+        DatabaseReference myRef = firebaseInstance.getReference("recipe");
+        myRef.child(Integer.toString(id)).setValue(recipe);
```

Comment:
Works well, but maybe we could use the `uid` of the recipe directly instead? This way we wouldn't need to read the id before pushing to the database.
Alternatively, we could use `push()` which create a unique key for the database.

Reply:
Will change this later